## **Mini-Project #2: Using Pandas Library specifically pandas.DataFrame.loc and pandas.DataFrame.iloc for analyzing [Trending Youtube Video Statistics](https://www.kaggle.com/datasets/datasnaek/youtube)**


---

Welcome to the second mini-project I built based on practical concept I learned about for using [**Pandas**](https://pandas.pydata.org/) library, specifically using [**pandas.DataFrame.loc**](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.loc.html) and [**pandas.DataFrame.iloc**](https://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.DataFrame.iloc.html) for analyzing [**Trending Youtube Video Statistics**](https://www.kaggle.com/datasets/datasnaek/youtube).

This jupyter notebook consist of all the concepts I've learned about using the specific method (**pandas.DataFrame.loc** and **pandas.DataFrame.iloc**) for filtering and manipulating rows and columns of a dataset. For this project, we will use the [**Trending Youtube Video Statistics**](https://www.kaggle.com/datasets/datasnaek/youtube) from **Kaggle**.

Specifically, it analyzes the trending youtube videos from **United States of America** that is accumulated from several months (and counting) of daily trending YouTube videos using the method that we've learned for filtering rows.

### **1. Loading dataset**

First, we need to import the `pandas` library using **Python** and load the dataset using `pd.read_csv()` method from our environment. We cannot analyze the data if we did not load it into our own environment.

In [11]:
import pandas as pd

USvideos = pd.read_csv("../datasets/USvideos.csv")

### **2. Data Overview**

Second, we need to inspect the dataset to check if it consist of messy or unstructured data such as **missing values**, **inconsistent formats**, **wrong datatypes**, **duplicate rows**, and etc. We need to clean it first to provide an accurate and detailed analysis based on our hypothesis.

In [12]:
USvideos.info()

<class 'pandas.DataFrame'>
RangeIndex: 40949 entries, 0 to 40948
Data columns (total 16 columns):
 #   Column                  Non-Null Count  Dtype
---  ------                  --------------  -----
 0   video_id                40949 non-null  str  
 1   trending_date           40949 non-null  str  
 2   title                   40949 non-null  str  
 3   channel_title           40949 non-null  str  
 4   category_id             40949 non-null  int64
 5   publish_time            40949 non-null  str  
 6   tags                    40949 non-null  str  
 7   views                   40949 non-null  int64
 8   likes                   40949 non-null  int64
 9   dislikes                40949 non-null  int64
 10  comment_count           40949 non-null  int64
 11  thumbnail_link          40949 non-null  str  
 12  comments_disabled       40949 non-null  bool 
 13  ratings_disabled        40949 non-null  bool 
 14  video_error_or_removed  40949 non-null  bool 
 15  description             40379 

Based on the output of the code cell above, we can see that the column `description` is the only column with missing values because the non-null count it have are `40,379`, which means only `40,379` rows have a value and some few rows has a missing value. But this was expected because some **YouTube Videos** does not provide description to their video so most likely we will just **provide a fix value to all missing values of that column**.

Also, some columns are in the wrong datatype such as `trending_date` which is in `str` datatype but it should be in `date` object and `publish_time` which is also in `str` datatype but it should be in `time` object. So we will fix it by **standardizing their datatypes**.

In [22]:
USvideos[["views", "likes", "dislikes", "comment_count"]].describe()

,views,likes,dislikes,comment_count
count,4.094900e+04,4.094900e+04,4.094900e+04,4.094900e+04
mean,2.360785e+06,7.426670e+04,3.711401e+03,8.446804e+03
std,7.394114e+06,2.288853e+05,2.902971e+04,3.743049e+04
min,5.490000e+02,0.000000e+00,0.000000e+00,0.000000e+00
25%,2.423290e+05,5.424000e+03,2.020000e+02,6.140000e+02
50%,6.818610e+05,1.809100e+04,6.310000e+02,1.856000e+03
75%,1.823157e+06,5.541700e+04,1.938000e+03,5.755000e+03
max,2.252119e+08,5.613827e+06,1.674420e+06,1.361580e+06


As we can see above, the standard deviation of every engagement metrics are a bit high and that is because that's the nature of social media metrics especially **YouTube** that every videos produce different number of `views`, `likes`, `dislikes`, and `comment_count` that's why some numbers produce outliers or consist of low engagement videos.